File: clients/java/src/main/java/cb/bots/MyAI.java
package cb.bots;

import cb.PlayState;

public class MyAI implements BotAi {
    private int column = 0;
    private int myCoinId = -1;

    public MyAI() {
    }

    @Override
    public int play(PlayState playState) {
        if (myCoinId == -1) {
            myCoinId = playState.getCoinId();
        }
        //  todo: put your logic code here
        return column;
    }

    @Override
    public String getName() {
        //  todo: give your bot a super duper cool name
        return "YourBotName";
    }
}

File: clients/kotlin/src/main/kotlin/cb/fourconnect/bots/MyAI.kt
package cb.fourconnect.bots

import cb.WebsocketClient
import cb.fourconnect.game.GameData
import cb.fourconnect.game.Response

class MyAI : FourConnectAI {
    // todo: give your bot a super duper cool name
    override val name: String = "MyAI"
    var coinId: Int = -1

    override fun play(gamePacket: GameData): Response {
        if (coinId == -1) {
            coinId = gamePacket.coinId
        }

        // todo: put your logic code here
        return Response(0)
    }
}

fun main() {
    val ai = MyAI()
    WebsocketClient().connect<GameData, Response>(ai = ai, path = ai.name)
}